packages/backend/server/src/base/config/loader.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable, Provider } from '@nestjs/common';
+import { merge } from 'lodash-es';
+
+import { getDefaultConfig } from './register';
+
+@Injectable()
+export class ConfigLoader {
+  static withOverrides(overrides: DeepPartial<AppConfig>): Provider {
+    return {
+      provide: ConfigLoader,
+      useFactory: () => new ConfigLoader(overrides),
+      inject: [],
+    };
+  }
+
+  #config: DeepReadonly<AppConfig> = this.loadDefault();
+
+  constructor(private readonly overrides: DeepPartial<AppConfig>) {}
+
+  get config() {
+    return this.#config;
+  }
+
+  override(updates: DeepPartial<AppConfig>) {
+    // TODO: do the validations
+    this.#config = Object.freeze(merge({}, this.#config, updates));
```

Comment:
Should it only be merged into the `this.#config` object instance? Instead of creating a new object instance? Otherwise, the Config that has already been injected will still reference the old config instance?

```ts
merge(this.#config, updates)
```